### ISSUE TITLE
pausableBuffered

### ISSUE THREAD

**Issue by mkko** [RxSwiftCommunity/RxSwiftExt#77]
### Name and description
`pausableBuffered` – pauses emission of values and emits the latest value on resume

### Motivation for inclusion

There already exists a `pausable` operator, but it doesn't store the latest value. This operator already exists in [RxJS](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/pausablebuffered.md) and simply for the sake of conformity it would be nice to have it here as well.

Sometimes the need is not to silence an observable (what `pausable`) but resume listening when ready and at that point it really makes sense to get the latest value.

### Example of use

An example from my personal need, where `pausable` didn't cut was displaying search results but only after they were ready to be presented (as this had side effects in the UI). Basically I wanted to perform the search proactively, but have the results only when a modal sheet was closed. Might not be the best example though, but I think this really has its place.

```
let isOverlayVisible = Variable(true)
let searchResults = searchInput
    .debounce(0.3, scheduler: MainScheduler.instance)
    .distinctUntilChanged()
    .flatMapLatest { str in searchService.search(str) }

let visibleSearchResults = searchResults.pausableBuffered(pauser: self.isOverlayVisible.asObservable())

isOverlayVisible.value = false
```

**Comment by mkko** [comment]
The solution is simple, I think. However, right now I don't have the time to go in to testing and documenting and such:

```
public func pausableBuffered<P : ObservableType>(pauser: P) -> Observable<E> where P.E == Bool {
    return Observable.combineLatest(self, pauser.distinctUntilChanged()) { ($0, $1) }
        .filter { (element, paused) in paused }
        .map { (element, paused) in element }
}
```

**Comment by fpillet** [comment]
Thank you! 👍

**Comment by mkko** [comment]
Awesome work, thank you! Especially @tgyhlsb.
